### PR TITLE
workflow: add release pipe-line

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
push firmware artifact and will ready to pre-release, if you push tag.

### CODE REVIEW

**Comment by RuffaloLavoisier** [comment]
How to test..?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Rename config
+        run: mv include/config.h.example include/config.h
+      - name: Compile language ${{ matrix.language }} model ${{ matrix.model }}
+        run: python3 .github/buildFirmware.py -l "${{ matrix.language }}" -m "${{ matrix.model }}" -b "${{ matrix.build-configuration }}"
+      - name: Upload firmware artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: firmwares
+          path: |
+            *.bin
```
Comment (simonmicro) [inline .github/workflows/release.yml:90]: Can we merge this into the existing jobs, please? Maybe you can use the `if: ` key to skip the publish if not needed.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  publish:
+    needs: build-OSW
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+      # create the GitHub release
+      - name: Create release
+        id: create_release
+        uses: actions/create-release@v1.1.4
```
Comment (simonmicro) [inline .github/workflows/release.yml:100]: While this runs on the tag, what will the pre-filled release-text be?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: "${{ github.ref }}"
+          draft: false
```
Comment (simonmicro) [inline .github/workflows/release.yml:106]: Maybe we could still make a draft? Who would be the owner of the release? The person, who triggered this pipeline (so the tag-pusher)?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+          prerelease: true
+      # upload assets
+      - name: Upload OSW-OS Firmware (zip)
+        uses: actions/upload-release-asset@v1.0.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (simonmicro) [inline .github/workflows/release.yml:112]: Is this default, or do we need to add a new secret for that?

**Comment by simonmicro** [comment]
> How to test..?

Maybe we can create a `v.0.16.3-rc1` and remove it later again?


Alternative: Fork this repo and run in on your side and see if it works.

**Comment by RuffaloLavoisier** [comment]
> Alternative: Fork this repo and run in on your side and see if it works.

Good idea.

File: .github/workflows/release.yml (hunk 3 above)
Comment (RuffaloLavoisier) [reply .github/workflows/release.yml:106]: > Maybe we could still make a draft?

yes, we can.
and default name , `github-action`.

File: .github/workflows/release.yml (hunk 2 above)
Comment (RuffaloLavoisier) [reply .github/workflows/release.yml:100]: Did you ok ? See my new answer with the test picture.

File: .github/workflows/release.yml (hunk 1 above)
Comment (RuffaloLavoisier) [reply .github/workflows/release.yml:90]: maybe we get attachment from other action. right?

File: .github/workflows/release.yml (hunk 4 above)
Comment (RuffaloLavoisier) [reply .github/workflows/release.yml:112]: let me see..

**Comment by RuffaloLavoisier** [comment]
i ii re-pull request...change my forked branch

**Pull request closed without merging**